packages/grid/x-data-grid-generator/src/useFakeServer.ts

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { GridRowModel } from '@mui/x-data-grid';
+
+function sleep(time: number): Promise<void> {
+  return new Promise<void>((res) => {
+    setTimeout(res, time);
+  });
+}
+
+export type FakeServerReturnType = {
+  get: any;
+};
+
+export const useFakeServer = (rows: GridRowModel[]): FakeServerReturnType => {
+  const get = React.useCallback(
+    async ({ page = 0, pageSize = 10, cursor }) => {
```

Comment:
Maybe it would be safer to not provide a default value to `page` to force the use of the API to be very explicit.

```suggestion
    async ({ page, pageSize = 10, cursor }) => {
```